tests/test_pytypes.py

```diff
@@ -225,6 +225,18 @@ def test_constructors():
     for k in noconv2:
         assert noconv2[k] is expected[k]
 
+    type_error_tests = [
```

Comment:
Maybe a separate `def test_nonconverting_constructor()`?
IIUC you're not depending on anything in `test_constructores`.

Reply:
That's true, yes, but it does fit under the name `test_constructors`.
I'm not too happy with these tests either way, to be fair :-/ So any suggestion for improvement is welcome.

Reply:
> That's true, yes, but it does fit under the name `test_constructors`.
> I'm not too happy with these tests either way, to be fair :-/ So any suggestion for improvement is welcome.

In general, the more granular the tests, the better (because independent failures don't block other tests). Combine them only if you have a good reason. In this particular case, I'd definitely put the new test in a separate function.